src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -74,7 +74,15 @@ internal string SanitizeName(string s, bool typeName = false)
                 // TODO: We assume that there won't be collisions with our own or C++ built-in identifiers.
                 sb.Append("_");
             }
-            return (sb != null) ? sb.ToString() : s;
+
+            string santizedName = (sb != null) ? sb.ToString() : s;
+
+            // The character sequences denoting generic instantiations, arrays, byrefs, or pointers must be
+            // restricted to that use only. Replace them if they happened to be used in any identifiers in 
+            // the compilation input.
+            return _mangleForCplusPlus
+                ? santizedName.Replace(EnterNameScopeSequence, "_AA_").Replace(ExitNameScopeSequence, "_VV_")
```

Comment:
(Or use `<` and `>` for when we're writing an object file, and the weird characters for C++ - basically I'm trying to see if we can get rid of the double `String.Replace`. If you look at the GC allocation profile of the compiler right now, we allocate a ton of strings because of how inefficient `NameMangler` already is)